util/dvsim/dvsim.py

```diff
@@ -75,14 +74,8 @@ def resolve_scratch_root(arg_scratch_root):
     else:
         arg_scratch_root = os.path.realpath(arg_scratch_root)
 
-    try:
-        os.system("mkdir -p " + arg_scratch_root)
-    except OSError:
-        log.fatal(
-            "Invalid --scratch-root=\"%s\" switch - failed to create directory!",
-            arg_scratch_root)
-        sys.exit(1)
-    return (arg_scratch_root)
+    os.makedirs(arg_scratch_root, exist_ok=True)
```

Comment:
Do you want to handle the case when I give `--scratch-root=/root` with a nice error message here?